semeio/jobs/scripts/fm_pyscal.py

```diff
@@ -0,0 +1,201 @@
+        # should be used for GasOil as well.
+        do_interpolation = True
+        int_param_go_name = int_param_wo_name
+    elif int_param_wo_name == MAGIC_NONE and int_param_go_name == MAGIC_NONE:
+        do_interpolation = False
+    else:
+        # Something is wrong if we end here
+        _logger.error("WaterOil interpolation parameter missing")
+        sys.exit(1)
+
+    slgof = slgof.lower()
+    if slgof not in ["sgof", "slgof"]:
+        _logger.error("Only supports sgof or slgof")
+        sys.exit(1)
+
+    if family not in [1, 2]:
+        _logger.error("Family must be either 1 or 2")
+        sys.exit(1)
+
+    if not do_interpolation:
+        int_param_wo = None
+        int_param_go = None
+    if do_interpolation:
+        # Read all key-value pairs from parameters.txt
+        if not os.path.exists(parameters_file_name):
+            _logger.error("%s does not exists", parameters_file_name)
+            sys.exit(1)
+        with open(parameters_file_name) as parameters_file:
+            parameters = parameters_file.readlines()
+        parameter_dict = rm_genkw_prefix(extract_key_value(parameters))
+
+        # Add some magic parameters:
+        parameter_dict.update(MAGIC_CASES)
+        if int_param_wo_name not in parameter_dict:
+            _logger.error(
+                "Requested parameter name %s not found in %s",
+                int_param_wo_name,
+                parameters_file_name,
+            )
+            sys.exit(1)
+        int_param_wo = float(parameter_dict[int_param_wo_name])
+        _logger.info(
+            "Collected %s from parameter.txt with value %f",
+            int_param_wo_name,
+            int_param_wo,
+        )
+        int_param_go = int_param_wo
+
+        if int_param_go_name not in parameter_dict:
+            _logger.error(
+                "Requested parameter name %s not found in %s",
+                int_param_go_name,
+                parameters_file_name,
+            )
+            sys.exit(1)
+        int_param_go = float(parameter_dict[int_param_go_name])
+        if int_param_go_name != int_param_wo_name:
+            _logger.info(
+                "Collected %s from parameter.txt with value %f",
+                int_param_go_name,
+                int_param_go,
+            )
+
+    if sheet_name in ["0", MAGIC_NONE]:
+        # Limitation: If the user names an xls sheet with the string "0"
+        # and it is not the first sheet, it will not be accessible with this
+        # forward model
+        sheet_name = None
+
+    try:
+        pyscalcli.pyscal_main(
+            parametertable=relperm_parameters_file,
+            verbose=True,
+            output=output_filename,
+            sheet_name=sheet_name,
+            int_param_wo=int_param_wo,
+            int_param_go=int_param_go,
+            slgof=slgof == "slgof",
+            family2=family == 2,
+        )
+    except ValueError as e_msg:
+        _logger.error(str(e_msg))
+        sys.exit(1)
```

Comment:
I would suggest that you do `raise e_msg` instead. Then you still get the message of the logger on `stdout`, but in `stderr` you will still get the original stack trace. Which is extremely useful if you need to debug something :)

Reply:
Fixed